### ISSUE TITLE
Vector128 invalid code generation with tiered compilation in Release

### ISSUE THREAD

**Issue by xoofx** [dotnet/runtime#96381]
### Description

Hey, I discovered a bug with tiered compilation that is generating an invalid code when a `Vector128<double>` is embedded in a struct and we perform calculation with it.

### Reproduction Steps

Create a program with the following code, `dotnet run -c Release`

```c#
using System.Runtime.Intrinsics;
using System.Runtime.CompilerServices;

var rect = new PseudoPointF64(1.0, 2.0);
var result = rect * 2.0;

Console.WriteLine(result);

public readonly struct PseudoPointF64
{
    private readonly Vector128<double> _value;

    public PseudoPointF64(Vector128<double> value)
    {
        _value = value;
    }

    public PseudoPointF64(double x, double y)
    {
        _value = Vector128.Create(x, y);
    }

    [MethodImpl(MethodImplOptions.AggressiveInlining)]

    public static PseudoPointF64 operator *(PseudoPointF64 left, double right) => new(left._value * Vector128.Create(right));


    public override string ToString()
    {
        return _value.ToString();
    }
}
```

### Expected behavior

It should print `<2, 4>`

### Actual behavior

It prints `<2, 0>`

### Known Workarounds

Add `[MethodImpl(MethodImplOptions.AggressiveOptimizations)]` to the method is fixing the issue.

### Configuration

* dotnet --version 8.0.100
* x64
* Windows 11 Pro
* Version	23H2
* OS build	22631.2861

**Comment by EgorBo** [comment]
Thanks for the minimal repro! Looks like it's already fixed in .NET 9.0 so we need to find the right commit.

**Comment by EgorBo** [comment]
Looks like https://github.com/dotnet/runtime/pull/90556

**Comment by xoofx** [comment]
I can back port it, if possible. What is the policy for backporting bugfixes like this?

**Comment by EgorBo** [comment]
> I can back port it, if possible. What is the policy for backporting bugfixes like this?

We have a built-in command to do backports 🙂
I think the fix should meet the bar for backports to 8.0 since it was reported by a customer (you) and it's of low risk, just needs an approval from the tactics.

**Comment by EgorBo** [comment]
The fix is backported to 8.0.2 and merged, closing the issue
